### ISSUE TITLE
Postgresql DB Engine Error: '>=' not supported between instances of 'datetime.timedelta' and 'int'

### ISSUE THREAD

**Issue by angelagilhotra** [apache/superset#15768]
There seems to be a problem in retrieving and using any column that is of the format: ` TIMESTAMP WITH TIME ZONE` or just ` TIMESTAMP` 

A relational table with a column of type ` TIMESTAMP WITH TIME ZONE` would error when run a `SELECT * ...` or `SELECT <column_with_timestamp_type> FROM...`

The SQL editor errors even with `SELECT now()`

The error is: 

```
DB engine Error

postgresql error: '>=' not supported between instances of 'datetime.timedelta' and 'int'

This may be triggered by:
Issue 1002 - The database returned an unexpected error.
```

### Expected results

Would expect the column to return properly and be usable in exploration

### Actual results & Screenshots

![image](https://user-images.githubusercontent.com/10832414/126075557-c5ec7815-4621-4fbb-8d0e-e220cd9dfd09.png)

#### How to reproduce the bug

1. Go to SQL Lab
2. run the query: `SELECT now()`

### Environment

my requirements.txt: 

```
aiohttp==3.7.4.post0
alembic==1.6.5
amqp==2.6.1
apache-superset==1.2.0
apispec==3.3.2
async-timeout==3.0.1
attrs==21.2.0
Babel==2.9.1
backoff==1.11.1
billiard==3.6.4.0
bleach==3.3.1
Brotli==1.0.9
cachelib==0.1.1
celery==4.4.7
cffi==1.14.6
chardet==4.0.0
click==7.1.2
colorama==0.4.4
contextlib2==21.6.0
convertdate==2.3.2
cron-descriptor==1.2.24
croniter==1.0.15
cryptography==3.4.7
decorator==5.0.9
defusedxml==0.7.1
dnspython==2.1.0
email-validator==1.1.3
Flask==1.1.4
Flask-AppBuilder==3.3.1
Flask-Babel==1.0.0
Flask-Caching==1.10.1
Flask-Compress==1.10.1
Flask-JWT-Extended==3.25.1
Flask-Login==0.4.1
Flask-Migrate==3.0.1
Flask-OpenID==1.2.5
Flask-SQLAlchemy==2.5.1
flask-talisman==0.8.1
Flask-WTF==0.14.3
geographiclib==1.52
geopy==2.2.0
graphlib-backport==1.0.3
gunicorn==20.0.4
holidays==0.10.3
humanize==3.10.0
idna==3.2
isodate==0.6.0
itsdangerous==1.1.0
Jinja2==2.11.3
jsonschema==3.2.0
kombu==4.6.11
korean-lunar-calendar==0.2.1
Mako==1.1.4
Markdown==3.3.4
MarkupSafe==2.0.1
marshmallow==3.12.2
marshmallow-enum==1.5.1
marshmallow-sqlalchemy==0.23.1
msgpack==1.0.2
multidict==5.1.0
numpy==1.21.0
packaging==21.0
pandas==1.2.5
parsedatetime==2.6
pathlib2==2.3.6
pgsanity==0.2.9
polyline==1.4.0
prison==0.1.3
psycopg2==2.9.1
py==1.10.0
pyarrow==3.0.0
pycparser==2.20
PyJWT==1.7.1
PyMeeus==0.5.11
pyparsing==2.4.7
pyrsistent==0.18.0
python-dateutil==2.8.2
python-dotenv==0.18.0
python-editor==1.0.4
python-geohash==0.8.5
python3-openid==3.2.0
pytz==2021.1
PyYAML==5.4.1
redis==3.5.3
retry==0.9.2
selenium==3.141.0
simplejson==3.17.3
six==1.16.0
slackclient==2.5.0
SQLAlchemy==1.3.24
SQLAlchemy-Utils==0.36.8
sqlparse==0.3.0
typing-extensions==3.10.0.0
urllib3==1.26.6
vine==1.3.0
webencodings==0.5.1
Werkzeug==1.0.1
WTForms==2.3.3
WTForms-JSON==0.3.3
yarl==1.6.3
```

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by zhaoyongjie** [comment]
Hi @nglglhtr.
I can't reproduce this issue on the latest master branch.

![image](https://user-images.githubusercontent.com/2016594/126091709-0dcbd8e8-3c91-4dd3-bb85-e5de0b1c9a78.png)

**Comment by srinify** [comment]
Hi @nglglhtr do you mind listing some reproduction instructions here, ideally using the default `examples` datasets that are provided?

**Comment by srinify** [comment]
Hi @nglglhtr I'm not able to reproduce this issue:

<img width="621" alt="Screen Shot 2021-07-19 at 10 29 05 AM" src="https://user-images.githubusercontent.com/801507/126176619-5f9c616d-44b9-4ba0-b9ab-8a27f27c1306.png">

Can you add some more details reproduction instructions / share a video ideally using the default `examples` datasets?

**Comment by wildsonc** [comment]
I have two instances, but only one have this issue
![image](https://user-images.githubusercontent.com/72408418/126213028-018b5ee2-fb30-40bc-b783-f1c77afe7155.png)

**Comment by srinify** [comment]
This will be hard to debug / fix if we can't reproduce :( Can you share more context?

Are you doing any `>=` comparison in the superset semantic layer anywhere? (e.g. metrics, calculated columns, etc).

**Comment by wildsonc** [comment]
i have 2 superset's connected in same database.

**Comment by wildsonc** [comment]
I have 2 superset's connected in same database.

Node 1:
![image](https://user-images.githubusercontent.com/72408418/126220581-8262d568-db2f-435f-b553-4c8c156e76d8.png)

Node 2:
![image](https://user-images.githubusercontent.com/72408418/126220597-df26bbd1-0560-4efa-a1da-c9af4b97d1a1.png)
Query error:
`Query 56: <class 'TypeError'>
2021-07-19 17:06:23,715:ERROR:superset.sql_lab:Query 56: <class 'TypeError'>
postgresql error: '>=' not supported between instances of 'datetime.timedelta' and 'int'
2021-07-19 17:06:23,840:WARNING:superset.views.base:postgresql error: '>=' not supported between instances of 'datetime.timedelta' and 'int'`

This error appears when there is a column of type date/timestamp, when I remove this column it works

**Comment by zhaoyongjie** [comment]
Hi @wildsonc Do you mind share your config file?

**Comment by wildsonc** [comment]
[superset_conifg.txt](https://github.com/apache/superset/files/6848889/superset_conifg.txt)

The error occurs in this connected database
![image](https://user-images.githubusercontent.com/72408418/126338001-e3d6f03c-1cd6-41af-832b-a65f3d65e1c2.png)

URI SQLALCHEMY: postgresql+psycopg2://explorernet:XXXXXXXXXX@10.9.35.41:5432/web_caixas
EXTRA: {
    "metadata_params": {},
    "engine_params": {},
    "metadata_cache_timeout": {},
    "schemas_allowed_for_csv_upload": []
}

No other options are checked in connection settings.

**Comment by nglglhtr** [comment]
@srinify 
> Are you doing any >= comparison in the superset semantic layer anywhere? (e.g. metrics, calculated columns, etc).

Not really, I'm simply trying to query a database that consists of a column of type timestamp

**Comment by nglglhtr** [comment]
Some more context: 

- the docker image doesn't have this error, which is what I used for local debugging and trying
- the `pip install superset` is what i used for my heroku deployment - which is what gives this error

**Comment by soitgoes511** [comment]
Same issue here. Docker instance had no issues with connecting to local postgres db but my local install of apache-superset installed via pip in venv yielded the following error despite successful connection w/ postgresql instance:

    DB engine Error
    postgresql error: '>=' not supported between instances of 'datetime.timedelta' and 'int'
    
    See more

And here is the query (very simple w/o any relational operators):

```sql
SELECT *
FROM pulse_ox_moving_average;
```
Here are the only packages installed in the venv, apart from apache-superset, I installed psycopg2 and that is all.

[requirements.txt](https://github.com/apache/superset/files/6873551/requirements.txt)
`
I do have one timestamped column, but this was a non-issue w/ the docker-compose install.

**Comment by soitgoes511** [comment]
After further experimentation, the issue seems to be timezones. If I removed the timezone offset with to_char(), my query works.

**Comment by Ruframapi** [comment]
I downgraded the version of psycopg2 and psycopg2-binary to 2.8.6 and the problem was fixed. I assume there is a problem with version 2.9.X.

**Comment by soitgoes511** [comment]
That did the trick. Very much appreciated. :heavy_check_mark:

**Comment by amrishparmar** [comment]
Dowgrading from 2.9.1 -> 2.8.6 was also the solution in my case.

Could quite possibly be related to this change listed in the [2.9 release notes](https://www.psycopg.org/docs/news.html#what-s-new-in-psycopg-2-9)? 
> Use datetime.timezone objects by default in datetime objects instead of FixedOffsetTimezone

**Comment by mail2lawi** [comment]
Following on your suggestion, I fixed it by editing the date column expression in the dataset editor window:

![image](https://user-images.githubusercontent.com/29949457/129150752-51d3a959-29a7-471a-b665-13fc718c5012.png)


> After further experimentation, the issue seems to be timezones. If I removed the timezone offset with to_char(), my query works.
> 
> **Update:**
> 
> I was able to work around the issue with the following query:
> 
> ```sql
> SELECT (time at time zone 'CEST')::timestamp without time zone AS "time",
>        ma_spo2,
>        ma_bpm,
>        ma_perf
> FROM pulse_ox_moving_average
> ```
> 
> ![superser_error_workaround](https://user-images.githubusercontent.com/22125622/126900290-2bd45d90-2389-44b2-9f59-2d12c0d9d197.png)
> 
> This hack would allow me to create time-series charts where as using to_char would not. Setting a timezone and then localizing by removing the timezone is not ideal for the long term.

**Comment by jonathanStrange0** [comment]
I can replicate this issue using a fresh install of the latest docker image per the instructions [here](https://superset.apache.org/docs/installation/installing-superset-using-docker-compose). 

Screenshot below.

![superset_issue_15768](https://user-images.githubusercontent.com/44637027/130059441-b59e7cfa-42d6-4304-9bba-8d2084a02415.png)

According to a pip freeze in the app container (superset_app), this version is using psycopg2 v. 2.9.1.

Manually replacing version 2.9.1 with 2.8.6 as suggested above resolves the issue.

@mail2lawi's solution above (time at time zone 'CEST')... also was an effective fix, leading me to support the hypothesis that this issue is specifically related to columns which contain timestamptz values.

Thank you for your help all!

**Comment by dsidlo** [comment]
I  am getting this error on the current master branch...

> commit bc4b6f0a6cfb05f0d913a5e4d2dc09844960e04a (HEAD -> master, origin/master, origin/HEAD)
> Author: John Bodley <4567245+john-bodley@users.noreply.github.com>
> Date:   Mon Aug 23 10:20:52 2021 -0700

I am connecting to postgresql 11.10.

Here is what I see in my logs...

> superset_app            | 2021-08-23 18:17:13,022:INFO:superset.views.core:Triggering query_id: 34
> superset_app            | Query 34: Executing 1 statement(s)
> superset_app            | 2021-08-23 18:17:13,052:INFO:superset.sql_lab:Query 34: Executing 1 statement(s)
> superset_app            | Query 34: Set query to 'running'
> superset_app            | 2021-08-23 18:17:13,052:INFO:superset.sql_lab:Query 34: Set query to 'running'
> superset_app            | Query 34: Running statement 1 out of 1
> superset_app            | 2021-08-23 18:17:13,177:INFO:superset.sql_lab:Query 34: Running statement 1 out of 1
> superset_app            | Query 34: <class 'TypeError'>
> superset_app            | Traceback (most recent call last):
> superset_app            |   File "/app/superset/sql_lab.py", line 276, in execute_sql_statement
> superset_app            |     data = db_engine_spec.fetch_data(cursor, increased_limit)
> superset_app            |   File "/app/superset/db_engine_specs/postgres.py", line 174, in fetch_data
> superset_app            |     return super().fetch_data(cursor, limit)
> superset_app            |   File "/app/superset/db_engine_specs/base.py", line 510, in fetch_data
> superset_app            |     raise cls.get_dbapi_mapped_exception(ex)
> superset_app            |   File "/app/superset/db_engine_specs/base.py", line 508, in fetch_data
> superset_app            |     return cursor.fetchall()
> superset_app            |   File "/usr/local/lib/python3.7/site-packages/pytz/__init__.py", line 403, in __init__
> superset_app            |     if abs(minutes) >= 1440:
> superset_app            | TypeError: '>=' not supported between instances of 'datetime.timedelta' and 'int'
> superset_app            | 2021-08-23 18:17:13,201:ERROR:superset.sql_lab:Query 34: <class 'TypeError'>
> superset_app            | Traceback (most recent call last):
> superset_app            |   File "/app/superset/sql_lab.py", line 276, in execute_sql_statement
> superset_app            |     data = db_engine_spec.fetch_data(cursor, increased_limit)
> superset_app            |   File "/app/superset/db_engine_specs/postgres.py", line 174, in fetch_data
> superset_app            |     return super().fetch_data(cursor, limit)
> superset_app            |   File "/app/superset/db_engine_specs/base.py", line 510, in fetch_data
> superset_app            |     raise cls.get_dbapi_mapped_exception(ex)
> superset_app            |   File "/app/superset/db_engine_specs/base.py", line 508, in fetch_data
> superset_app            |     return cursor.fetchall()
> superset_app            |   File "/usr/local/lib/python3.7/site-packages/pytz/__init__.py", line 403, in __init__
> superset_app            |     if abs(minutes) >= 1440:
> superset_app            | TypeError: '>=' not supported between instances of 'datetime.timedelta' and 'int'
> superset_app            | [SupersetError(message="postgresql error: '>=' not supported between instances of 'datetime.timedelta' and 'int'", error_type=<SupersetErrorType.GENERIC_DB_ENGINE_ERROR: 'GENERIC_DB_ENGINE_ERROR'>, level=<ErrorLevel.ERROR: 'error'>, extra={'engine_name': 'PostgreSQL', 'issue_codes': [{'code': 1002, 'message': 'Issue 1002 - The database returned an unexpected error.'}]})]
> superset_app            | Traceback (most recent call last):
> superset_app            |   File "/app/superset/views/base.py", line 204, in wraps
> superset_app            |     return f(self, *args, **kwargs)
> superset_app            |   File "/app/superset/utils/log.py", line 242, in wrapper
> superset_app            |     value = f(*args, **kwargs)
> superset_app            |   File "/app/superset/views/core.py", line 2578, in sql_json
> superset_app            |     return self.sql_json_exec(request.json, log_params)
> superset_app            |   File "/app/superset/views/core.py", line 2767, in sql_json_exec
> superset_app            |     session, rendered_query, query, expand_data, log_params
> superset_app            |   File "/app/superset/views/core.py", line 2563, in _sql_json_sync
> superset_app            |     [SupersetError(**params) for params in data["errors"]]
> superset_app            | superset.exceptions.SupersetErrorsException: [SupersetError(message="postgresql error: '>=' not supported between instances of 'datetime.timedelta' and 'int'", error_type=<SupersetErrorType.GENERIC_DB_ENGINE_ERROR: 'GENERIC_DB_ENGINE_ERROR'>, level=<ErrorLevel.ERROR: 'error'>, extra={'engine_name': 'PostgreSQL', 'issue_codes': [{'code': 1002, 'message': 'Issue 1002 - The database returned an unexpected error.'}]})]
> superset_app            | 2021-08-23 18:17:13,219:WARNING:superset.views.base:[SupersetError(message="postgresql error: '>=' not supported between instances of 'datetime.timedelta' and 'int'", error_type=<SupersetErrorType.GENERIC_DB_ENGINE_ERROR: 'GENERIC_DB_ENGINE_ERROR'>, level=<ErrorLevel.ERROR: 'error'>, extra={'engine_name': 'PostgreSQL', 'issue_codes': [{'code': 1002, 'message': 'Issue 1002 - The database returned an unexpected error.'}]})]
> superset_app            | Traceback (most recent call last):
> superset_app            |   File "/app/superset/views/base.py", line 204, in wraps
> superset_app            |     return f(self, *args, **kwargs)
> superset_app            |   File "/app/superset/utils/log.py", line 242, in wrapper
> superset_app            |     value = f(*args, **kwargs)
> superset_app            |   File "/app/superset/views/core.py", line 2578, in sql_json
> superset_app            |     return self.sql_json_exec(request.json, log_params)
> superset_app            |   File "/app/superset/views/core.py", line 2767, in sql_json_exec
> superset_app            |     session, rendered_query, query, expand_data, log_params
> superset_app            |   File "/app/superset/views/core.py", line 2563, in _sql_json_sync
> superset_app            |     [SupersetError(**params) for params in data["errors"]]
> superset_app            | superset.exceptions.SupersetErrorsException: [SupersetError(message="postgresql error: '>=' not supported between instances of 'datetime.timedelta' and 'int'", error_type=<SupersetErrorType.GENERIC_DB_ENGINE_ERROR: 'GENERIC_DB_ENGINE_ERROR'>, level=<ErrorLevel.ERROR: 'error'>, extra={'engine_name': 'PostgreSQL', 'issue_codes': [{'code': 1002, 'message': 'Issue 1002 - The database returned an unexpected error.'}]})]
>

**Comment by dsidlo** [comment]
The error seems to be related the following code...

> superset_app            |   File "/usr/local/lib/python3.7/site-packages/pytz/__init__.py", line 403, in __init__
> superset_app            |     if abs(minutes) >= 1440:
> superset_app            | TypeError: '>=' not supported between instances of 'datetime.timedelta' and 'int'
> superset_app            | [SupersetError(message="postgresql error: '>=' not supported between instances of 'datetime.timedelta' and 'int'", error_type=<SupersetErrorType.GENERIC_DB_ENGINE_ERROR: 'GENERIC_DB_ENGINE_ERROR'>, level=<ErrorLevel.ERROR: 'error'>, extra={'engine_name': 'PostgreSQL', 'issue_codes': [{'code': 1002, 'message': 'Issue 1002 - The database returned an unexpected error.'}]})]

The minutes variable is a datetime.timedelta object, which is not compatible with the '>=' operator against and int, in python.

**Comment by santicomp2014** [comment]
> with 2.8.6 as suggested ab

Hi @jonathanStrange0 , I'm new to superset.
How do you change the version, I cloned latest version and found `["psycopg2-binary==2.8.5"` as a dependency.
This version appears to be older than 2.8.6 and I get the same error.

What are the steps to change the version with docker-compose.
Thank you very much.

**Comment by dsidlo** [comment]
> As a work-around I convert the date to a string.
> So instead of...
> 
> ` select rfq._created_dt, ...`
> 
> use...
> 
> ` select to_char(rfq._created_dt, 'YYY-MM-DD') as _created_dt, ...`
> 
> The actual error does not seem to come from the Postgres server as indicated by the error message.
> I did some debugging, and it turns out that in the condition...
> `if abs(minutes) >= 1440:`
> ... minutes is a datetime.timedelta object which can't be evaluated against and int using the '>=' operator.
> But exception shows up as if it were coming from postgres.

String solution does not work well, it will cause other problems with some visualizations. The real issue is that dates can be missing the timezone in some way, and forcing a timezone into the output field fixes the date issue properly without making it a string. Making the main temporal date field into a string, will cause other problems.

Don't use...
`select date_fld,  ...`
Use...
`select date_fld::timestamptz at time zone 'EST',  `

**Comment by frafra** [comment]
Same problem happens when using "Big Number with Trendline" and doing a `COUNT(*)` over a timestamp column. Using Docker image `apache/superset:4dc859f89e5668ed3f94cd1ef0532a301a3ab85a`

**Comment by lsyarn** [comment]
Work fine  after modify postgres db engine, and timezone offset problem in explore solved.
Steps with docker-compose instance(version=1.3.1).
```bash
docker exec -it superset_app /bin/bash
apt-get update
apt-get install vim
vim /app/superset/db_engine_specs/postgres.py 
# delete line 171 (cursor.tzinfo_factory = FixedOffsetTimezone)
exit
# back to host machine
docker-compose -f docker-compose-non-dev.yml restart
```
[psycopg 2.9 release note](https://www.psycopg.org/docs/news.html#what-s-new-in-psycopg-2-9)
psycopg2 updated to use `datetime.timezone` as `tz` property of datetime object, we do not need `pytz._FixedOffset` any more.

**Comment by cclements128** [comment]
Is this an issue that will be fixed soon?
In its current state, Superset charts and datasets don't work with any Postgresql columns of type "timestamp with time zone".

**Comment by jsanko9** [comment]
Maybe label should be changed ?  Feel like this is being forgotten because of that.
Additionally @Isyarn may have even found a fix already. 
Unfortunately, I don't feel competent enough to create PR.

**Comment by raivtash** [comment]
> Work fine after modify postgres db engine, and timezone offset problem in explore solved. Steps with docker-compose instance(version=1.3.1).
> 
> ```shell
> docker exec -it superset_app /bin/bash
> apt-get update
> apt-get install vim
> vim /app/superset/db_engine_specs/postgres.py 
> # delete line 171 (cursor.tzinfo_factory = FixedOffsetTimezone)
> exit
> # back to host machine
> docker-compose -f docker-compose-non-dev.yml restart
> ```
> 
> [psycopg 2.9 release note](https://www.psycopg.org/docs/news.html#what-s-new-in-psycopg-2-9) psycopg2 updated to use `datetime.timezone` as `tz` property of datetime object, we do not need `pytz._FixedOffset` any more.

This fix did not work for me. But reverting psycopg2-binary to 2.8.5 solved my problem.

**Comment by pvergain** [comment]
> Work fine after modify postgres db engine, and timezone offset problem in explore solved. Steps with docker-compose instance(version=1.3.1).
> 
> ```shell
> docker exec -it superset_app /bin/bash
> apt-get update
> apt-get install vim
> vim /app/superset/db_engine_specs/postgres.py 
> # delete line 171 (cursor.tzinfo_factory = FixedOffsetTimezone)
> exit
> # back to host machine
> docker-compose -f docker-compose-non-dev.yml restart
> ```
> 
> [psycopg 2.9 release note](https://www.psycopg.org/docs/news.html#what-s-new-in-psycopg-2-9) psycopg2 updated to use `datetime.timezone` as `tz` property of datetime object, we do not need `pytz._FixedOffset` any more.


This solution works for me.

![image](https://user-images.githubusercontent.com/57228/144882866-3f1d753e-bc2b-4e01-a017-5686e799d0e2.png)

**Comment by pvergain** [comment]
> > I can replicate this issue using a fresh install of the latest docker image per the instructions [here](https://superset.apache.org/docs/installation/installing-superset-using-docker-compose).
> > Screenshot below.
> > ![superset_issue_15768](https://user-images.githubusercontent.com/44637027/130059441-b59e7cfa-42d6-4304-9bba-8d2084a02415.png)
> > According to a pip freeze in the app container (superset_app), this version is using psycopg2 v. 2.9.1.
> > Manually replacing version 2.9.1 with 2.8.6 as suggested above resolves the issue.
> > @mail2lawi's solution above (time at time zone 'CEST')... also was an effective fix, leading me to support the hypothesis that this issue is specifically related to columns which contain timestamptz values.
> > Thank you for your help all!
> 
> Hi @jonathanStrange0 , I'm new to superset. How do you change the version, I cloned master and found `["psycopg2-binary==2.8.5"` as a dependency. This version appears to be older than 2.8.6 and I get the same error.
> 
> What are the steps to change the version with docker-compose. Thank you very much.

Hi !

For docker-compose-non-dev, here is my "hack":

```
    docker-compose -f docker-compose-non-dev.yml up
    docker-compose -f docker-compose-non-dev.yml exec superset pip uninstall -y psycopg2-binary
    docker-compose -f docker-compose-non-dev.yml exec superset pip install psycopg2-binary==2.8.6
    docker-compose -f docker-compose-non-dev.yml restart superset
```

Before the hack
==============

![image](https://user-images.githubusercontent.com/57228/145030199-b5b3e03e-e45f-4196-9a29-15ea70c73fb2.png)

After the hack
==============

![image](https://user-images.githubusercontent.com/57228/145030265-0a2c9f15-8d7f-47a6-bca3-cfd4cb5ea064.png)

**Comment by pvergain** [comment]
> @santicomp2014,
> 
> You should just be able to:
> 
>     * Clone the repo:  $ git clone https://github.com/apache/superset.git
> 
>     * $ cd superset/requirements/
> 
>     * Change both **docker.txt** and **development.txt** to _psycopg2-binary==2.8.6_
> 
>     * Run: $ docker-compose -f docker-compose-non-dev.yml up

For me, this didn't work. I found this "hack" : https://github.com/apache/superset/issues/15768#issuecomment-987887089

```
    docker-compose -f docker-compose-non-dev.yml up
    docker-compose -f docker-compose-non-dev.yml exec superset pip uninstall -y psycopg2-binary
    docker-compose -f docker-compose-non-dev.yml exec superset pip install psycopg2-binary==2.8.6
    docker-compose -f docker-compose-non-dev.yml restart superset
```

**Comment by poxeron** [comment]
This library (psycopg2-binary) was not installed at all. 

1) pip uninstall -y psycopg2-binary
2) pip install psycopg2-binary==2.8.6

**Comment by ronaldvanrij** [comment]
Glad to see there are workarounds possible, however I'd prefer to use the official Docker image posted to Dockerhub. Any idea when this fix can be made available there?

**Comment by basaravikiran** [comment]
I also had the same issue and downgrading psycopg2 from 2.9.1 to 2.8.6 did fix it.

**Comment by villebro** [comment]
This has been fixed on master branch here: #17713
